src/main/java/io/cryostat/core/reports/InterruptibleReportGenerator.java

```diff
@@ -336,43 +338,95 @@ private List<HtmlResultGroup> loadResultGroups(Element element) {
+                        // access entries in the ResultProvider
+                        if (shouldEvaluate(evaluatedRules, rule)) {
+                            IResult result;
+                            logger.trace("Processing rule {}", rule.getName());
+                            ReportRuleEvalEvent evt = new ReportRuleEvalEvent(rule.getName());
+                            evt.begin();
+                            try {
+                                // Check that the rule has all of the events it needs to evaluate
+                                // first
+                                if (!RulesToolkit.matchesEventAvailabilityMap(
+                                        items, rule.getRequiredEvents())) {
+                                    logger.warn(
+                                            "Rule missing required events: {} ", rule.getName());
+                                    result =
+                                            ResultBuilder.createFor(
+                                                            rule,
+                                                            IPreferenceValueProvider.DEFAULT_VALUES)
+                                                    .setSeverity(Severity.NA)
+                                                    .build();
+                                } else {
+                                    resultFuture.run();
+                                    result = resultFuture.get();
+                                }
+                                evaluatedRules.put(rule.getClass(), result.getSeverity());
+                                rp.addResults(result);
+                                return result;
+                            } finally {
+                                evt.end();
+                                if (evt.shouldCommit()) {
+                                    evt.commit();
+                                }
                             }
+                        } else {
+                            logger.warn("Rule is missing dependencies: {} ", rule.getName());
+                            return ResultBuilder.createFor(
+                                            rule, IPreferenceValueProvider.DEFAULT_VALUES)
+                                    .setSeverity(Severity.NA)
+                                    .build();
                         }
                     });
         }
         return callables;
     }
 
+    /** Brought over from org.openjdk.jmc.flightrecorder.rules.jdk.test.TestRulesWithJFR */
+    private static boolean shouldEvaluate(
+            Map<Class<? extends IRule>, Severity> evaluatedRules, IRule rule) {
+        DependsOn dependency = rule.getClass().getAnnotation(DependsOn.class);
+        if (dependency != null) {
+            Class<? extends IRule> dependencyType = dependency.value();
+            if (dependencyType != null) {
+                while (true) {
```

Comment:
What is this `while (true)` needed for? If `evaluatedRules.containsKey(dependencyType)` is `false` then it looks like this just gets stuck spinning, and if it's `true` then it only runs once and returns immediately.

Reply:
That loop isn't needed in our case, it looks like that loop is only used/needed for the tests in jmc core, I missed that RulesToolkit has it's own version, I'll update it